packages/pages/src/common/src/parsers/sourceFileParser.ts

```diff
@@ -207,4 +208,115 @@ export default class SourceFileParser {
   getAllText(): string {
     return this.sourceFile.getFullText();
   }
+
+  /**
+   * For example, we can do getVariablePropertyByType("ModuleConfig", "name")
+   * and recieve the string value that the user set as name for their
+   * variable of type ModuleConfig
+   *
+   * @param type of the variable
+   * @param property of the variable
+   * @returns property of the variable as any
+   */
+  getVariablePropertyByType(type: string, property: string): any {
+    const declaration = this.getVariableDeclarationByType(type);
+    if (declaration === undefined) {
+      return;
+    }
+    const variableType = declaration.getType();
+    if (variableType.getText().includes(type)) {
+      const initializer = declaration.getInitializer();
+      if (
+        initializer &&
+        initializer.getKind() === SyntaxKind.ObjectLiteralExpression
+      ) {
+        const objectLiteral = initializer as any;
+        const variableProperty = objectLiteral
+          .getProperties()
+          .find(
+            (prop: { getName: () => string }) => prop.getName() === property
+          );
+        if (variableProperty) {
+          const value = variableProperty
+            .getFirstChildByKind(SyntaxKind.StringLiteral)
+            ?.getText();
+          if (value) {
+            return value;
+          } else {
+            return variableProperty;
+          }
+        } else {
+          console.log(type + " does not have a '" + property + "' property.");
```

Comment:
Do we just console.log in other places or do we error? I think we'd want to error out and fail the build.